examples/with-eip-1193-provider/src/components/dashboard.tsx

```diff
@@ -0,0 +1,193 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  parseEther,
+  Address as AddressType,
+  formatEther,
+  getAddress,
+  Hex,
+  numberToHex,
+  parseGwei,
+} from "viem";
+
+import { TurnkeyEIP1193Provider } from "@turnkey/eip-1193-provider";
+import Address from "./address";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { sepolia } from "viem/chains";
+import { toast } from "sonner";
+// import { ToastAction } from '@/components/ui/toast';
+
+type DashboardProps = {
+  provider: TurnkeyEIP1193Provider;
+};
+
+export function Dashboard({ provider }: DashboardProps) {
+  const [balance, setBalance] = useState<bigint | null>(null);
+  const [selectedAccount, setSelectedAccount] = useState<AddressType | "">("");
+  const [sendToAddress, setSendToAddress] = useState<AddressType | "">("");
+  const [sendAmount, setSendAmount] = useState("");
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchAccountAndBalance = async () => {
+      if (provider) {
+        const accounts = await provider.request({ method: "eth_accounts" });
+        setSelectedAccount(accounts[0]);
+
+        const balanceInWei = await provider.request({
+          method: "eth_getBalance",
+          params: [accounts[0], "latest"],
+        });
+        const balanceInEther = BigInt(balanceInWei);
+        setBalance(balanceInEther);
+      }
+    };
+
+    fetchAccountAndBalance();
+  }, [provider]);
+
+  const handleSendTransaction = async () => {
+    if (provider && selectedAccount && sendToAddress && sendAmount) {
+      const chainId = numberToHex(sepolia.id);
+      const nonce = await provider.request({
+        method: "eth_getTransactionCount",
+        params: [selectedAccount, "latest"],
+      });
+      const gas = numberToHex(21000);
+      const maxFeePerGas = numberToHex(parseGwei("20"));
```

Comment:
tiny nit: we may need to do some live estimation here given sepolia has times of high traffic (relatively rare, but it does happen)

in our demo passkey wallet, we simply double estimates to be safe: https://github.com/tkhq/demo-passkey-wallet/blob/2182c36583fbda79f762ce5d0c70db4926feb547/internal/ethereum/ethereum.go#L61-L63